cpp/src/cider/exec/nextgen/context/CodegenContext.h

```diff
@@ -179,23 +164,19 @@ class CodegenContext {
     int64_t ctx_id;
     std::string name;
     // TODO(qiuyang) : LinearProbeHashTable will provide the default template
-    LinearProbeHashTable<int, std::pair<Batch*, int64_t>, murmurHash, Equal>* hash_table;
+    cider::exec::processor::JoinHashTable* hash_table;
 
     HashTableDescriptor(
         int64_t id,
         const std::string& n,
         // TODO(qiuyang) : LinearProbeHashTable will provide the default template
-        LinearProbeHashTable<int, std::pair<Batch*, int64_t>, murmurHash, Equal>* table =
-            new LinearProbeHashTable<int, std::pair<Batch*, int64_t>, murmurHash, Equal>(
-                16,
-                0))
+        cider::exec::processor::JoinHashTable* table =
+            new cider::exec::processor::JoinHashTable())
         : ctx_id(id), name(n), hash_table(table) {}
   };
 
-  void setHashTable(
-      LinearProbeHashTable<int, std::pair<Batch*, int64_t>, murmurHash, Equal>&
-          LP_hash_table) {
-    hashtable_descriptor_.first->hash_table = &LP_hash_table;
+  void setHashTable(cider::exec::processor::JoinHashTable& join_hash_table) {
+    hashtable_descriptor_.first->hash_table = &join_hash_table;
```

Comment:
I am afraid a reference don't have address.
https://stackoverflow.com/questions/1950779/is-there-any-way-to-find-the-address-of-a-reference